network/p2p/handler.go

```diff
@@ -51,8 +51,8 @@ type responder struct {
 }
 
 func (r *responder) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID uint32, deadline time.Time, request []byte) error {
-	appResponse, err := r.handler.AppRequest(ctx, nodeID, deadline, request)
-	if err != nil {
+	appResponse := r.handler.AppRequest(ctx, nodeID, deadline, request)
+	if len(appResponse) == 0 {
```

Comment:
should we update the log to something like `no response` instead of `failed to handle` or do you think `failed to handle` is the right message here?

Reply:
Basically wondering if an empty response should always indicate failure to handle

Reply:
So app requests always need a corresponding app response, or else we fire the app response failed handler. In that sense, if you require a response, you should always be using app request, and if you don't require a response, I believe you should be using app gossip instead. The only difference between the two messages is that request requires a response (no response means we also register you as having been unresponsive, etc), whereas gossip makes no such guarantee.

Reply:
Updated this to be "dropping empty response"